src/Mindee/Extraction/ImageExtractor.cs

```diff
@@ -0,0 +1,319 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using Docnet.Core;
+using Docnet.Core.Models;
+using Mindee.Exceptions;
+using Mindee.Geometry;
+using Mindee.Input;
+using Mindee.Parsing.Standard;
+using SkiaSharp;
+
+namespace Mindee.Extraction
+{
+    /// <summary>
+    /// Extract sub-images from an image.
+    /// </summary>
+    public class ImageExtractor
+    {
+        private readonly List<SKBitmap> _pageImages;
+        private readonly string _filename;
+        private readonly string _saveFormat;
+
+        /// <summary>
+        /// Init from a Local Input Source.
+        /// </summary>
+        /// <param name="localInput"></param>
+        public ImageExtractor(LocalInputSource localInput)
+        {
+            this._filename = localInput.Filename;
+            this._pageImages = new List<SKBitmap>();
+            if (localInput.IsPdf())
+            {
+                this._saveFormat = "jpg";
+                List<SKBitmap> pdfPageImages = PdfToImages(localInput.FileBytes);
+                this._pageImages.AddRange(pdfPageImages);
+            }
+            else
+            {
+                this._pageImages.Add(SKBitmap.Decode(localInput.FileBytes));
+                var extension = Path.GetExtension(localInput.Filename);
+                this._saveFormat = extension == null ? ".jpg" : extension.Substring(1);
+            }
+        }
+
+        /// <summary>
+        /// Init from a path.
+        /// </summary>
+        /// <param name="filePath">Path to the file.</param>
+        public ImageExtractor(string filePath)
+        {
+            this._filename = filePath;
+            this._pageImages = new List<SKBitmap>();
+
+            if (IsPdf(filePath))
+            {
+                this._saveFormat = "jpg";
+                List<SKBitmap> pdfPageImages = PdfToImages(File.ReadAllBytes(filePath));
+                this._pageImages.AddRange(pdfPageImages);
+            }
+            else
+            {
+                string[] splitName = SplitNameStrict(filePath);
+                this._saveFormat = splitName[1].ToLower();
+
+                using FileStream fs = new FileStream(filePath, FileMode.Open, FileAccess.Read);
+                using SKManagedStream skStream = new SKManagedStream(fs);
+                this._pageImages.Add(SKBitmap.Decode(skStream));
+            }
+        }
```

Comment:
init a `LocalInputSource` from the filePath and call the `public ImageExtractor(LocalInputSource localInput)` constructor.

Reply:
I think you meant to tag the test where I failed to do so instead of this?

Reply:
It's a good thing to be able to construct from a string path, just you need to call the other constructor, not duplicate the code.

Reply:
I wanted to avoid having to create a LocalInputSource the user did not have access to.

Reply:
I'll just add it as an accessible property I think